### ISSUE TITLE
Model answer "x" gives "CAS failed to return any data. ans1:x"

### ISSUE THREAD

**Issue by christianp** [maths/moodle-qtype_stack#191]
I've just installed STACK from github and tried to create a question. I set the model answer for ans1 to `x`. When I try to save, I get an error message "CAS failed to return any data. ans1:x".
If I put a more long-winded model answer in, like `x+0`, or the quick start guide's suggestion of `3*(x-1)^2`, I can save the question.
I've installed Maxima 5.36.1 from source.

Sorry if this isn't a bug and I should know better, I'm trying to do something else and this just tripped me up so I thought I'd note it down.

**Comment by christianp** [comment]
When I preview the question with `x+0` as the model answer, I can type `x` as my answer and get full marks, and in fact that's what the advice field suggests.

**Comment by sangwinc** [comment]
Thanks Christian,

I can't reproduce this problem.
Did you cut and paste the "x" from somewhere with possible hidden HTML tags?
Can you reproduce the problem on a separate question?

Chris

**Comment by christianp** [comment]
Here's my question, with the model answer set to `+x` so I could save it. My only interaction with the page after clicking "Create a new question" was to type (not paste) `x` in the model answer field for ans1, and `ans1` and `x` in the SAns and TAns fields of prt1.
How would I get more debug information about what the CAS is doing when I try to save?

[questions-stuff-Default for stuff-20160223-1114.zip](https://github.com/maths/moodle-qtype_stack/files/142344/questions-stuff-Default.for.stuff-20160223-1114.zip)

**Comment by sangwinc** [comment]
Christian,

I'm totally baffled by this.  I can't reproduce the problem on either Moodle 2.7 or Moodle 3.0.

The problem could well be elsewhere in your question.

This is the line which looks for errors from the CAS.
https://github.com/maths/moodle-qtype_stack/blob/master/edit_stack_form.php#L1422

I've just pushed changes to the master branch which enable you to see the raw input-output connection to the CAS in cases where the connection totally fails.   This will be a significant help, reflecting back on my previous similar bugs.  You need to have "qtype_stack | casdebugging" set to Yes in your qtype_stack configuration page.

What do you get now?

**Comment by christianp** [comment]
Clearing the CAS cache fixed it. I first installed Maxima from the Ubuntu repository, which had a version that didn't work, so I reinstalled from source. STACK must have cached a result with the old version of Maxima. For reference, here's the error I got with the cached result.

![stack cas cache error](https://cloud.githubusercontent.com/assets/19513/13279878/80d788e4-dad2-11e5-8207-a1a901efa7cd.png)

A way of preventing this in future might be to store the Maxima version with each cached result, or to clear the cache when the Maxima version changes. I don't know how the cache works, so I don't know if that's reasonable.
